src/glcm.jl

```diff
@@ -0,0 +1,154 @@
+function glcm{T<:Real}(img::AbstractArray{T, 2}, distance::Integer, angle::Real, mat_size::Integer = 16)
+	max_img = maximum(img)
+	img_rescaled = map(i -> max(1, Int(ceil(i * mat_size / max_img))), img)
+	_glcm(img_rescaled, distance, angle, mat_size)
+end
+
+function glcm{T<:Real, A<:Real}(img::AbstractArray{T, 2}, distances::Array{Int, 1}, angles::Array{A, 1}, mat_size::Integer = 16)
+	max_img = maximum(img)
+	img_rescaled = map(i -> max(1, Int(ceil(i * mat_size / max_img))), img)
+	glcm_matrices = [_glcm(img_rescaled, d, a, mat_size) for d in distances, a in angles]
+	glcm_matrices
+end
+
+function glcm{T<:Colorant}(img::AbstractArray{T, 2}, distance::Integer, angle::Real, mat_size::Integer = 16)
+	img_rescaled = map(i -> max(1, Int(ceil((convert(Gray{U8}, i).val.i) * mat_size / 256))), img)
+	_glcm(img_rescaled, distance, angle, mat_size)
+end
+
+function glcm{T<:Colorant, A<:Real}(img::AbstractArray{T, 2}, distances::Array{Int, 1}, angles::Array{A, 1}, mat_size::Integer = 16)
+	img_rescaled = map(i -> max(1, Int(ceil((convert(Gray{U8}, i).val.i) * mat_size / 256))), img)	
+	glcm_matrices = [_glcm(img_rescaled, d, a, mat_size) for d in distances, a in angles]
+	glcm_matrices
+end
+
+glcm{T<:Union{Colorant, Real}, A<:Real}(img::AbstractArray{T, 2}, distances::Int, angles::Array{A, 1}, mat_size::Integer = 16) = glcm(img, [distances], angles, mat_size)
+glcm{T<:Union{Colorant, Real}}(img::AbstractArray{T, 2}, distances::Array{Int, 1}, angles::Real, mat_size::Integer = 16) = glcm(img, distances, [angles], mat_size)
+
+function _glcm{T}(img::AbstractArray{T, 2}, distance::Integer, angle::Number, mat_size::Integer)
+	co_oc_matrix = zeros(Integer, mat_size, mat_size)
+
+	R = CartesianRange(size(img))
+
+	for I in R
+
+	int_one = img[I]
+
+	co_occuring_pixel = I + CartesianIndex(Int(round(sin(angle) * distance)), Int(round(cos(angle) * distance)))
```

Comment:
You're not getting a lot of benefit from `CartesianIndex` here because your algorithm is explicitly two-dimensional. If it only makes sense as a 2d algorithm, then it might be easier to just write this as `for j = 1:size(img,2), i = 1:size(img,1)`...